### ISSUE TITLE
fromJSON('""') results in error on Windows

### ISSUE THREAD

**Issue by wch** [jeroen/jsonlite#180]
On a Mac, it works fine:

```
> jsonlite::fromJSON('""')
[1] ""
```

On Windows, it gives an error:

```
> jsonlite::fromJSON('""')
Warning in open.connection(con, "rb") :
  cannot open file '""': Invalid argument
Error in open.connection(con, "rb") : cannot open the connection
```

I believe it's because `file.exists('""')` returns TRUE on Windows: https://github.com/jeroen/jsonlite/blob/013a348/R/fromJSON.R#L91

**Comment by wch** [comment]
I've filed a bug for R here: https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17265

But it should probably be worked around in here as well.

**Comment by jeroen** [comment]
Pfff, is this a new bug in R 3.4?

**Comment by wch** [comment]
I actually tested it only in R 3.3.3.

**Comment by jeroen** [comment]
Any suggestion for a good workaround that would not have side effects for the case of strange filenames?

The alternative is to export the json parser function directly (which is not overloaded for files and urls).

**Comment by jeroen** [comment]
@wch I think this should do the job: https://github.com/jeroen/jsonlite/commit/aa5a6ea447993d674f7b33954a0e2dcd26daad2b. It only calls `file.exists()` if the string itslef is not json.

**Comment by wch** [comment]
I think that workaround makes sense. It's not perfect (people will get confusing error messages for strings like `fromJSON('"')`) but I think it's a reasonable for these relatively uncommon cases. And once the bug is fixed in R it'll become even less common. Hopefully there's not too much of a performance cost for running `validate()`.

**Comment by wch** [comment]
Update on the issue in R. It's been closed as a WONTFIX:
https://bugs.r-project.org/bugzilla/show_bug.cgi?id=17265

**Comment by jeroen** [comment]
Ill try tot do a release soon.

**Comment by jeroen** [comment]
FYI this has landed in `jsonlite 1.5` on CRAN.
